### PR TITLE
make EKS create, delete, update flow fully resilient

### PR DESCRIPTION
**Pull request by sancyx**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
- move create, delete, update EKS cluster code from cluster.EKSCluster to separate creator, deleter, updater
- the whole create update delete flow, including cluster setup, db persist, happens now in workflow
- move k8s config download code to workflow activity
- delete unused code from EKSCluster
- EKSCluster can be completely removed from cluster package once we can delete UpdateNodepools API endpoint 

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)

### CODE REVIEW

File: internal/providers/amazon/eks/driver/cluster_creator.go [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+	err = commonCluster.(*cluster.EKSCluster).SetCurrentWorkflowID(exec.GetID())
+	if err != nil {
+		return nil, err
+	}
+
+	timer, err := getClusterStatusChangeMetricTimer(commonCluster.GetCloud(), commonCluster.GetLocation(), pkgCluster.Creating, commonCluster.GetOrganizationId(), commonCluster.GetName(), c.statusChangeDurationMetric)
+	if err != nil {
+		return nil, err
+	}
+	go func() {
```
Comment (stoader) [inline internal/providers/amazon/eks/driver/cluster_creator.go]: This won't track the duration of an EKS cluster creation if Pipeline is restarted.

File: internal/providers/amazon/eks/driver/cluster_creator.go [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+				nodePools = append(nodePools, np)
+			}
+		}
+	}
+	return nodePools
+}
+
+// Create implements the clusterCreator interface.
+func (c *EksClusterCreator) create(ctx context.Context, commonCluster cluster.CommonCluster, createRequest *pkgCluster.CreateClusterRequest) (cluster.CommonCluster, error) {
+	c.logger.Info("start creating EKS Cluster")
```
Comment (stoader) [inline internal/providers/amazon/eks/driver/cluster_creator.go]: Add context to the logger such as `cluster` , `organizationID`, `correlationID` etc

File: internal/providers/amazon/eks/driver/cluster_creator.go [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+	}
+
+	timer, err := getClusterStatusChangeMetricTimer(commonCluster.GetCloud(), commonCluster.GetLocation(), pkgCluster.Creating, commonCluster.GetOrganizationId(), commonCluster.GetName(), c.statusChangeDurationMetric)
+	if err != nil {
+		return nil, err
+	}
+	go func() {
+		err = exec.Get(ctx, nil)
+		if err != nil {
+			c.logger.Errorf("Cluster create workflow failed: %v", err)
```
Comment (stoader) [inline internal/providers/amazon/eks/driver/cluster_creator.go]: Error messages should start with lower case

**Comment by bonifaido** [comment]
LGTM

**Pull request merged**